src/hooks/useCheckNickName.ts

```
@@ -0,0 +1,29 @@
+export function useCheckNickName() {
+    const [nickNameMessage, setNickNameMessage] = useState('');
+    const [nickNameAvailable, setNickNameAvailable] = useState(false);
+    const [isCheckingNickName, setIsCheckingNickName] = useState(false);
+
+    const handleCheckNickName = async (nickName: string) => {
+        setIsCheckingNickName(true);
+        try {
+            const response = await checkNickName(nickName);
+            setNickNameMessage(response.message || '사용 가능한 닉네임입니다.');
+            setNickNameAvailable(true);
+        } catch (error: unknown) {
```

ESLint

[error] 'error' is defined but never used.

(@typescript-eslint/no-unused-vars)